### PR TITLE
docs: update `tf.summary.text()`

### PR DESCRIPTION
**Pull request by bileschi**
Update documentation for `tf.summary.text`.

Adds some examples and examples of what features are supported.

#API_FIXIT_3_18_21

### CODE REVIEW

File: tensorboard/plugins/text/summary_v2.py
Comment (psybuzz) [inline tensorboard/plugins/text/summary_v2.py:29]: Maybe personal taste, but I think that seeing '.scalar, .SummaryWriter, .image' is a bit redundant with the left sidepane for navigation in the devsite, especially if all tf.summary.* pages will mention this.

wdyt about something more generic along the lines of: "See the rest of the `tf.summary.*` methods for ways to write other data types"?

imo, this "see also" might make more sense at the bottom, near the mention of the demo notebook.
wdyt?

File: tensorboard/plugins/text/summary_v2.py
Comment (psybuzz) [inline tensorboard/plugins/text/summary_v2.py:57]: Our docs aren't always consistent about writing "tensor" vs "\`Tensor\`", but I see the latter in many of the TF devsite docs.  Should we replace instances in this paragraph with "\`Tensor\`"?

File: tensorboard/plugins/text/summary_v2.py
Comment (psybuzz) [inline tensorboard/plugins/text/summary_v2.py:33]: Nice.  Much of this paragraph applies to tf.summary.image, so I will copy/paste if that is alright :)

**Review by psybuzz — APPROVED** [review]
overall lgtm, I will update the .image() PR to be more consistent with this.

we might want to wait for nfelt to lgtm before landing as well

File: tensorboard/plugins/text/summary_v2.py
Comment (psybuzz) [inline tensorboard/plugins/text/summary_v2.py:37]: nit: personally, I prefer shorter logdir like "logs" or "logdir".  wdyt?

File: tensorboard/plugins/text/summary_v2.py
Comment (bileschi) [reply tensorboard/plugins/text/summary_v2.py:29]: I put these here to be consistent with the ui pattern for TensorFlow's https://www.tensorflow.org/api_docs/python/tf/concat

Note that the See Also section is handled specially by the docsite software.

File: tensorboard/plugins/text/summary_v2.py
Comment (bileschi) [reply tensorboard/plugins/text/summary_v2.py:57]: done

File: tensorboard/plugins/text/summary_v2.py
Comment (bileschi) [reply tensorboard/plugins/text/summary_v2.py:37]: done

File: tensorboard/plugins/text/summary_v2.py
Comment (psybuzz) [reply tensorboard/plugins/text/summary_v2.py:29]: Gotcha, after seeing tf.concat's docs, this makes sense.  I'll also add similar "See also"s for .image, .histogram!

File: tensorboard/plugins/text/summary_v2.py
Comment (nfelt) [inline tensorboard/plugins/text/summary_v2.py:31]: Same comment as other PR re: mentioning default writer, consistency w.r.t. choice of leading verb.

File: tensorboard/plugins/text/summary_v2.py
Comment (nfelt) [inline tensorboard/plugins/text/summary_v2.py:61]: Ditto here again about whether we've checked a bare link works?

File: tensorboard/plugins/text/summary_v2.py
Comment (nfelt) [inline tensorboard/plugins/text/summary_v2.py:59]: Maybe add "to avoid this, reshape the Tensor to at most rank-2 prior to passing it to this function."?

**Review by nfelt — APPROVED** [review]
Thanks for doing this!

**Pull request merged**
